### ISSUE TITLE
[Discussion] Performance numbers

### ISSUE THREAD

**Issue by csherratt** [gfx-rs/gfx_scene#41]
I did a little bit of profiling with `gfx_scene` to get an idea of how well the library was performing. I had a pathologically bad use-case where I was supplying a list of 5000 culled objects to the library (with no culling).

Section | Time 
-----------| ------
Engine | 4.0ms
[Scene Enqueue](https://github.com/gfx-rs/gfx_scene/blob/c1bb89ca4fba5f809e09e8c601898b4bb789ec59/src/scene/cull.rs#L105-125) | 7.7ms
[Phase Sort](https://github.com/gfx-rs/gfx_scene/blob/c1bb89ca4fba5f809e09e8c601898b4bb789ec59/src/phase/phase.rs#L270) | 0.4ms
[Phase Draw](https://github.com/gfx-rs/gfx_scene/blob/c1bb89ca4fba5f809e09e8c601898b4bb789ec59/src/phase/phase.rs#L272-274) | 136.0ms
Present | 1.3ms

In summary, most of our time is spend calling `stream.draw`

**Comment by kvark** [comment]
Thanks for the numbers! The fact we are spending time in `stream.draw` means that `gfx_scene` itself is not a bottleneck of this case.

**Comment by csherratt** [comment]
@kvark yes that is what I wanted to make sure of. Secondly, if there is a bottlenect it is the Phase enqueue.

I'm getting inconsitent numbers about the cost of `stream.draw`. I'm investigating it right now. My laptop did not have any problems drawing the scene at 30fps, even with 5000 objects. Which is what I was hoping for.

**Comment by csherratt** [comment]
Since #31 is done, I am closing this issue. Anyone can now gather the same numbers.
